src/Math-Matrix/PMTensor.class.st

```diff
@@ -0,0 +1,152 @@
+Class {
+	#name : #PMTensor,
+	#superclass : #Object,
+	#instVars : [
+		'array',
+		'shape',
+		'first',
+		'strides'
+	],
+	#category : #'Math-Matrix'
+}
+
+{ #category : #'instance creation' }
+PMTensor class >> fromNestedArray: anArray [
+
+	^ self new array: anArray flattened withShape: (self shape:anArray)
+]
+
+{ #category : #'instance creation' }
+PMTensor class >> newWith: anInteger [
+
+	^ self new array: {anInteger} withShape: #(  )
+]
+
+{ #category : #accessing }
+PMTensor class >> shape: anArray [
+
+anArray isArray ifFalse:[^#()]
+ifTrue:[
+	^ {anArray size}, (self shape: anArray first)
+	]
+
+]
+
+{ #category : #comparing }
+PMTensor >> = aTensor [
+
+	^ array = aTensor array & (first = aTensor first)
+	  & (strides = aTensor strides) & (shape = aTensor shape)
+]
+
+{ #category : #accessing }
+PMTensor >> array [ 
+^array
+]
+
+{ #category : #'as yet unclassified' }
+PMTensor >> array: aFlatArray withShape: aShape [
+
+	array := aFlatArray.
+	shape := aShape copy.
+	self updateFirst.
+	shape ifNotEmpty: [ self updateStrides]
+]
+
+{ #category : #accessing }
+PMTensor >> first [ 
```

Comment:
I'd expect a method called `first` to be similar to `Collection >> first`, i.e. be equivalent to `at: 1`. Maybe the instance variable `first` should be renamed as well to get a better name for its accessor.

Reply:
first came from Numpy's naming but I agree this is colliding with Collection>>first Idea for a better name?